src/electrum.rs

```diff
@@ -275,6 +275,71 @@ impl Rpc {
         Ok(status)
     }
 
+    fn outpoint_subscribe(&self, (txid, vout): (Txid, u32)) -> Result<Value> {
+        let funding = OutPoint { txid, vout };
+
+        let funding_blockhash = self.tracker.get_blockhash_by_txid(funding.txid);
+        let spending_blockhash = self.tracker.get_blockhash_spending_by_outpoint(funding);
+
+        let mut funding_inputs_confirmed = true;
+
+        let funding_tx =  match self.daemon.get_transaction(&funding.txid, funding_blockhash) {
+            Ok(tx) => tx,
+            Err(_) => return Ok(json!({})),
```

Comment:
Here I return an empty JSON to follow the Electrum 1.5 spec.

`get_transaction` is defined here:
https://github.com/romanz/electrs/blob/d153bfa2eb4435fbe2f70d38ece76152c0bcff66/src/daemon.rs#L139-L147

There is a context defined in case of error. Is it sufficient ?

Reply:
Ah, `anyhow` is screwing us.  I think this should work:

let funding_tx =  match self.daemon.get_raw_transaction(&funding.txid, funding_blockhash.as_ref()) {
    Ok(tx) => tx,
    Err(bitcoincore_rpc::Error::JsonRpc(jsonrpc::Error::Rpc(jsonrpc::error::RpcError { code: -5, .. }))) => Ok(json!({})),
    Err(error) => Err(error),
}

Reply:
I found `anyhow::Context` weird to handle. I had to keep the same match but I add a new match in error case.

In the error case, I downcast `anyhow` to a `bitcoinrpc::Error` to remove the context and match its type as you did. If it does, I return the empty JSON, else I just return the error.

Is it the right way to do it ?

Reply:
I personally find it messy even though not wrong. That's why I suggested calling `get_raw_transaction` directly instead of via `get_transaction` which does the same thing except messing error type. If @romanz has a different opinion, I don't mind.

Reply:
Would calling `get_raw_transaction` require the rpc field of daemon to be made public ? I don't know what it would imply...

It looks like a matter of preferences here, I don't mind either haha

Reply:
> I don't know what it would imply...

AFAIK nobody uses `electrs` as a library so nothing substantial. I think whole `daemon` struct may be just relic of the past when there was manual implementation.

Thinking about it, maybe the correct way to do it would be to change `get_transaction` to return `Result<Option<Transaction>>` instead of `Result<Transaction>`  and then convert not found errors to `None` similarly to how e.g. `HashMap` works except fallible.